lib/Backends/OpenCL/OpenCL.h

```diff
@@ -82,19 +82,11 @@ class OpenCLFunction final : public CompiledFunction {
   };
   /// The IR to be executed.
   std::unique_ptr<IRFunction> F_;
-  /// CL compute device id.
-  cl_device_id deviceId_;
-  /// CL compute context.
-  cl_context context_;
-  /// CL compute command queue.
-  cl_command_queue commands_;
   /// Cache of compiled programs.
   /// The same source code can be compile with different options (e.g. with
   /// different set of macro definitions) and/or for a different device and
   /// would result in different programs.
   std::unordered_map<ProgramKey, cl_program, ProgramKeyHash> programsCache_;
```

Comment:
May be not for this PR, but `kernelLaunches_`  should be moved out of `OpenCLFunction` as it collects a per-run information.

And may be `programsCache_` should be moved out of `OpenCLFunction` as well.

cc @gcatron